cpp/oneapi/dal/io/csv/common.hpp

```diff
@@ -65,13 +68,19 @@ class ONEDAL_EXPORT data_source_base : public base {
 using v1::data_source_tag;
 using v1::data_source_impl;
 using v1::data_source_base;
+using v1::is_valid_float_v;
 
 } // namespace detail
 
 namespace v1 {
 
+template <typename Float = float>
 class data_source : public detail::data_source_base {
```

Comment:
Please add the description to this class and its methods.
We use doxygen notations. You can look [here](https://github.com/oneapi-src/oneDAL/blob/main/cpp/oneapi/dal/table/row_accessor.hpp) for an example of how the classes/methods descriptions look like in oneDAL.